lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -1740,3 +1762,215 @@ DomV8BreakIterator createV8BreakIterator() {
     ],
   );
 }
+
+@JS()
+@staticInterop
+class DOMPoint {}
+extension DOMPointExtension on DOMPoint {
+  external double get x;
+  external double get y;
+  external double get z;
+  external double get w;
+}
+
+@JS()
+@anonymous
+@staticInterop
+class DOMPointInit {
+  external factory DOMPointInit({
+    double x = 0,
+    double y = 0,
+    double z = 0,
+    double w = 1,
+  });
```

Comment:
I normally don't separate the `external factory` from the staticInterop class that it creates. I think you should be able to do this:

```dart
@JS()
@staticInterop
class DOMPoint {
  external factory DOMPoint({
    double x = 0,
    double y = 0,
    double z = 0,
    double w = 1,
  });
}

extension DOMPointExtension on DOMPoint {
  external double get x;
  external double get y;
  external double get z;
  external double get w;
}
```

(And then the types of stuff elsewhere can receive a `DOMPoint` and a `DOMMatrix` rather than the `...Init` versions?)